base/src/AffineTransform.cpp

```diff
@@ -0,0 +1,380 @@
+			double shx, shy;
+			shx = (1 - co) * (srcSize[0].width / 2) + si * srcSize[0].height / 2;
+			shy = (-si) * (srcSize[0].width / 2) + (1 - co) * srcSize[0].height / 2;
+			double acoeff[2][3] = { {co, -si, shx + props.x}, {si, co, shy + props.y} };
+			status = nppiWarpAffine_8u_C4R_Ctx(const_cast<const Npp8u *>(static_cast<Npp8u *>(buffer)),
+				srcSize[0],
+				srcPitch[0],
+				srcRect[0],
+				static_cast<Npp8u *>(outBuffer),
+				dstPitch[0],
+				dstRect[0],
+				acoeff,
+				NPPI_INTER_NN,
+				nppStreamCtx);
+		}
+
+		if (status != NPP_SUCCESS)
+		{
+			LOG_ERROR << "resize failed<" << status << ">";
+		}
+
+		return true;
+	}
+
+	void setProps(AffineTransformProps &mprops)
+	{
+		if (!mOutputMetadata.get())
+		{
+			return;
+		}
+		auto rawMetadata = FrameMetadataFactory::downcast<RawImageMetadata>(mOutputMetadata);
+		props = mprops;
+	}
+
+public:
+	size_t mFrameLength;
+	framemetadata_sp mOutputMetadata;
+	std::string mOutputPinId;
+	AffineTransformProps props;
+	bool setMetadataHelper(framemetadata_sp &input, framemetadata_sp &output)
+	{
+		if (mFrameType == FrameMetadata::RAW_IMAGE)
+		{
+			auto inputRawMetadata = FrameMetadataFactory::downcast<RawImageMetadata>(input);
+			auto outputRawMetadata = FrameMetadataFactory::downcast<RawImageMetadata>(output);
+
+			channels = inputRawMetadata->getChannels();
+			srcSize[0] = { inputRawMetadata->getWidth(), inputRawMetadata->getHeight() };
+			srcRect[0] = { 0, 0, inputRawMetadata->getWidth(), inputRawMetadata->getHeight() };
+			srcPitch[0] = static_cast<int>(inputRawMetadata->getStep());
+			srcNextPtrOffset[0] = 0;
+			dstSize[0] = { outputRawMetadata->getWidth(), outputRawMetadata->getHeight() };
+			dstRect[0] = { 0, 0, outputRawMetadata->getWidth(), outputRawMetadata->getHeight() };
+			dstPitch[0] = static_cast<int>(outputRawMetadata->getStep());
+			dstNextPtrOffset[0] = 0;
+		}
+		else if (mFrameType == FrameMetadata::RAW_IMAGE_PLANAR)
+		{
+			auto inputRawMetadata = FrameMetadataFactory::downcast<RawImagePlanarMetadata>(input);
+			auto outputRawMetadata = FrameMetadataFactory::downcast<RawImagePlanarMetadata>(output);
+
+			channels = inputRawMetadata->getChannels();
+
+			for (auto i = 0; i < channels; i++)
+			{
+				srcSize[i] = { inputRawMetadata->getWidth(i), inputRawMetadata->getHeight(i) };
+				srcRect[i] = { 0, 0, inputRawMetadata->getWidth(i), inputRawMetadata->getHeight(i) };
+				srcPitch[i] = static_cast<int>(inputRawMetadata->getStep(i));
+				srcNextPtrOffset[i] = inputRawMetadata->getNextPtrOffset(i);
+
+				dstSize[i] = { outputRawMetadata->getWidth(i), outputRawMetadata->getHeight(i) };
+				dstRect[i] = { 0, 0, outputRawMetadata->getWidth(i), outputRawMetadata->getHeight(i) };
+				dstPitch[i] = static_cast<int>(outputRawMetadata->getStep(i));
+				dstNextPtrOffset[i] = outputRawMetadata->getNextPtrOffset(i);
+			}
+		}
+		return true;
+	}
+
+	FrameMetadata::FrameType mFrameType;
+	int depth;
+	int channels;
+	NppiSize srcSize[4];
```

Comment:
Add comments for the variables that are tough to understand explaining in one line what they are for.